Sources/Redis/RedisCommandTarget.swift

```diff
@@ -91,6 +91,84 @@ public extension RedisCommandTarget { // Future based
     return _enqueue([ "KEYS", pattern ])
   }
   
+  func scan(cursor: String = "0", pattern: String?, count: Int? = nil)
+    -> EventLoopFuture<(String, [String])>
+  {
+    var req = [
+      "SCAN".toRESPValue(),
+      cursor.toRESPValue()
+    ]
+    
+    if let pattern = pattern {
+      req.append(contentsOf: [
+        "MATCH".toRESPValue(),
+        pattern.toRESPValue()
+      ])
+    }
+    
+    if let count = count {
+      req.append(contentsOf: [
+        "COUNT".toRESPValue(),
+        count.toRESPValue()
+      ])
+    }
+    
+    return _enqueue(req)
+      .map() { (resp: RESPValue) in
+        guard case .array(.some(let items)) = resp, items.count == 2,
+              let newCursor = items[0].stringValue,
+              case .array(.some(let keyValues)) = items[1]
+          else
+        {
+          fatalError("unexpected result")
```

Comment:
Not a big fan of fatal errors. W/o looking into the details I'd do some sensible defaults instead of crashing everything.